File: ABADiversity/src/app/entities/survey.ts
import {Certificates,CompanyProfiles,FirmDemographics,
    Firms,JoinedLawyers,LeadershipDemographics,
    LeftLawyers,PromotionsAssociatePartners,ReducedHoursLawyers,
    TopTenHighestCompensations,UndertakenInitiatives} from './entities'
    
export interface Survey{
    Firm?:Firms;
    Companyprofile?:CompanyProfiles;
    UndertakenInitiatives ?: UndertakenInitiatives;
    FirmDemographics ?: FirmDemographics[],
    JoinedLawyers ?: JoinedLawyers[],
    LeadershipDemographics ?: LeadershipDemographics[];
    LeftLawyers ?: LeftLawyers[];
    PromotionsAssociatePartners ?: PromotionsAssociatePartners[];
    ReducedhoursLawyers ?: ReducedHoursLawyers[];
    Certificates ?: Certificates[];
    TopTenHighestCompensations ?: TopTenHighestCompensations[];
    IsNewFirm?:boolean;
}
export interface Survey2{
    firm?:Firms;
    companyProfile?:CompanyProfiles;
    undertakenInitiatives ?: UndertakenInitiatives;
    firmDemographics ?: FirmDemographics[],
    joinedLawyers ?: JoinedLawyers[],
    leadershipDemographics ?: LeadershipDemographics[];
    leftLawyers ?: LeftLawyers[];
    promotionsAssociatePartners ?: PromotionsAssociatePartners[];
    reducedhoursLawyers ?: ReducedHoursLawyers[];
    certificates ?: Certificates[];
    topTenHighestCompensations ?: TopTenHighestCompensations[];
    isNewFirm?:boolean;
}

File: API/DTO/Survey.cs
using API.Tables;
using System.Collections.Generic;

namespace API.DTO
{
  public class Survey
  {
    public Firms Firm { get; set; }
    public CompanyProfiles CompanyProfile { get; set; }
    public UndertakenInitiatives UndertakenInitiatives { get; set; }
    public List<FirmDemographics> FirmDemographics { get; set; }
    public List<JoinedLawyers> JoinedLawyers { get; set; }
    public List<LeadershipDemographics> LeadershipDemographics { get; set; }
    public List<LeftLawyers> LeftLawyers { get; set; }
    public List<PromotionsAssociatePartners> PromotionsAssociatePartners { get; set; }
    public List<ReducedHoursLawyers> ReducedHoursLawyers { get; set; }
    public List<Certificates> Certificates { get; set; }
    public List<TopTenHighestCompensations> TopTenHighestCompensations { get; set; }
    public bool IsNewFIrm { get; set; }
  }
}
